### PR TITLE
@craigspaeth - Fetch random related artist

### PR DESCRIPTION
**Pull request by broskoski**
Sorry for the sloppy history, picking up something I've been tinkering with for a while.

Right now on the homepage, we fetch the last artist the user followed and get an related suggestion based on that. That works fine, but what would be better is if we fetched a different random artist / related pair every time. The tricky part here is that two requests have to match up:

1. On page load we get a response of all the modules that _should_ be displayed. The client should be able to render parts of these modules initially as placeholders (like the title).
2. Then on the client side we fetch each module again individually with all the results, depending on what keys and parameters are returned in the initial fetch. This means that the random artist / related pair needs to match up between the first fetch and the secondary fetch. So.....

The approach now is, if the `me/modules` returns for a user that we _should_ show a related artists rail (i.e. the user follows any artist), we hit the `user/${userID}/suggested/similar/artists`, get some results, pick out a random pair, and then drop it into the `params` object for the subsequent to use.

With me??????

cc @mzikherman @alloy 

p.s. Hi @alloy, I'm back again.

### CODE REVIEW

**Comment by broskoski** [comment]
Check out this WIP force PR for an idea of how we would use this on the client:
https://github.com/artsy/force/compare/master...broskoski:related-artists-redux?expand=1

File: schema/home/home_page_artwork_modules.js [hunk 1]
```diff
@@ -4,55 +4,79 @@ import {
...
+
+        // For the related artists rail, we need to fetch a random
+        // set of followed artist + related artist initially
+        // and pass it along so that any placeholder titles are consistent
+        const relatedArtistIndex = findIndex(filteredModules, { key: 'related_artists' });
+
+        if (relatedArtistIndex > -1) {
+          return Promise.resolve(relatedArtist(accessToken, userID))
+          .then(({ artist, sim_artist }) => {
+            console.log('artist', artist, 'sim_artist', sim_artist);
```
Comment (alloy) [inline schema/home/home_page_artwork_modules.js]: Intended?

**Comment by alloy** [comment]
Nice tests! 👌

**Comment by alloy** [comment]
I still wonder if it doesn’t make more sense, API design wise, to use the Global Object Identification interface for this. The ID could then encode these parameters instead of having to add [2 parameters to the artwork module field](https://github.com/artsy/metaphysics/pull/388/files#diff-d3db7c66d593b9f5202ed5b60c20bdddR81) that are only really used in the case of 1 type of artwork rail.

(Let me know if you need a more elaborate example.)

**Comment by alloy** [comment]
I’m just spit-balling btw, don’t let this hold-up any progress you need to make right now.

File: schema/home/home_page_artwork_modules.js (hunk 1 above)
Comment (broskoski) [inline schema/home/home_page_artwork_modules.js]: Oops!

File: schema/home/context.js [hunk 2]
```diff
@@ -80,13 +78,16 @@ export const moduleContext = {
       return assign({}, fair, { context_type: 'Fair' });
     });
   },
-  related_artists: ({ accessToken }) => {
-    return followedArtist(accessToken).then((follow) => {
-      return relatedArtist(accessToken).then((artist) => ({
+  related_artists: ({ params }) => {
+    return Promise.all([
```
Comment (craigspaeth) [inline schema/home/context.js]: 👌

File: schema/home/home_page_artwork_modules.js [hunk 3]
```diff
@@ -4,55 +4,77 @@ import {
...
-        featuredAuction(),
-        featuredFair(),
-      ]).then(([auction, fair]) => {
-        const modules = loggedOutModules(auction, fair);
-        return filteredModules(modules, max_rails);
+        const filteredModules = filterModules(modules, max_rails);
+
+        // For the related artists rail, we need to fetch a random
+        // set of followed artist + related artist initially
+        // and pass it along so that any placeholder titles are consistent
```
Comment (craigspaeth) [inline schema/home/home_page_artwork_modules.js]: :ok

**Comment by craigspaeth** [comment]
Thanks for explaining this irl to me! Quite the complex bit of orchestration logic here—could you imagine having to duplicate this across clients 😕 💭

**Comment by artsyit** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;R73TaKi7&#x2F;368-home-v1-5"> Home v1.5</a>

**Pull request merged**